### PR TITLE
Loki: Update labels in log browser when time range changes

### LINKED ISSUE #36278: Loki: Log browser doesn't respect new time range when fetching labels
**What would you like to be added**:
Loki log browser fetches the labels with initial time range and doesn't update labels when time range changes. It is using selected time range to show values of the labels. 

https://user-images.githubusercontent.com/30407135/123834521-ca7b2a80-d907-11eb-97d0-93b210de4279.mov

This can be problematic/confusing in many ways as at other places, we usually pass the the selected time range to queries.

**Why is this needed**:
To always show all available labels.

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
Loki log browser fetched the labels with initial time range and doesn't update labels when time range changes. This PR fixes t in the similar way as it is done in Prometheus metrics browser.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/36278

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiQueryField.tsx [hunk 1]
```diff
@@ -98,6 +92,18 @@ export class LokiQueryField extends React.PureComponent<LokiQueryFieldProps, Lok
...
 
+  async componentDidUpdate(prevProps: LokiQueryFieldProps) {
+    const {
+      range,
+      datasource: { languageProvider },
+    } = this.props;
+    const refreshLabels = shouldRefreshLabels(range, prevProps.range);
+    // We want to refresh labels when range changes (we round up intervals to a minute)
+    if (refreshLabels) {
+      await languageProvider.fetchLabels();
```
Comment (gabor) [inline public/app/plugins/datasource/loki/components/LokiQueryField.tsx:103]: nitpick: as this is the last command of the function, i think you can remove the "await" here and remove the "async" at the start of `componentDidUpdate`

in other words, there is no need to wait for the result of `languageProvider.fetchLabels`

**Review by gabor — APPROVED** [review]
code is fine (i one nitpick-comment left), works fine, LGTM!

File: public/app/plugins/datasource/loki/components/LokiQueryField.tsx [hunk 2]
```diff
@@ -98,6 +92,18 @@ export class LokiQueryField extends React.PureComponent<LokiQueryFieldProps, Lok
     this.setState({ labelsLoaded: true });
   }
 
+  async componentDidUpdate(prevProps: LokiQueryFieldProps) {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/LokiQueryField.tsx:95]:
```suggestion
  componentDidUpdate(prevProps: LokiQueryFieldProps) {
```

File: public/app/plugins/datasource/loki/components/LokiQueryField.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiQueryField.tsx:103]:
```suggestion
      languageProvider.fetchLabels();
```

**Pull request merged**